model_api/cpp/models/src/classification_model.cpp

```diff
@@ -358,6 +359,27 @@ std::unique_ptr<ResultBase> ClassificationModel::get_hierarchical_predictions(In
     return retVal;
 }
 
+ov::Tensor ClassificationModel::reorder_saliency_maps(const ov::Tensor& source_maps) {
+    if (!hierarchical || source_maps.get_shape().size() == 1) {
+        return source_maps;
+    }
+
+    auto reordered_maps = ov::Tensor(source_maps.get_element_type(), source_maps.get_shape());
+    const std::uint8_t* source_maps_ptr = static_cast<std::uint8_t*>(source_maps.data());
+    std::uint8_t* reordered_maps_ptr = static_cast<std::uint8_t*>(reordered_maps.data());
+
+    size_t shape_offset = (source_maps.get_shape().size() == 4) ? 1 : 0;
+    size_t map_byte_size = source_maps.get_element_type().size() *
+        source_maps.get_shape()[shape_offset + 1] * source_maps.get_shape()[shape_offset + 2];
+
+    for (size_t i = 0; i < source_maps.get_shape()[shape_offset]; ++i) {
```

Comment:
Does it work if there are several maps returned, so source_maps have batch size 2? For instance (2,3,7,7) for 2 images and 3 classes